limitador/benches/bench.rs

```diff
@@ -54,70 +52,77 @@ struct TestCallParams {
-            bench_update_counters(b, test_scenario, storage);
-        },
-        TEST_SCENARIOS.to_vec(),
-    );
-}
-
-fn bench_update_counters_redis(c: &mut Criterion) {
-    c.bench_function_over_inputs(
-        &format!("update_counters (redis)"),
-        |b: &mut Bencher, test_scenario: &&TestScenario| {
-            let storage = Box::new(RedisStorage::default());
-            bench_update_counters(b, test_scenario, storage);
-        },
-        TEST_SCENARIOS.to_vec(),
-    );
+impl Display for TestScenario {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{} namespaces with {} limits each with {} conditions and {} variables",
+            self.n_namespaces, self.n_limits_per_ns, self.n_conds_per_limit, self.n_vars_per_limit
+        )
+    }
 }
 
-fn bench_check_rate_limited_and_update_in_mem(c: &mut Criterion) {
-    c.bench_function_over_inputs(
-        &format!("check_rate_limited_and_update (in mem)"),
-        |b: &mut Bencher, test_scenario: &&TestScenario| {
-            let storage = Box::new(InMemoryStorage::default());
-            bench_check_rate_limited_and_update(b, test_scenario, storage);
-        },
-        TEST_SCENARIOS.to_vec(),
-    );
+fn bench_in_mem(c: &mut Criterion) {
+    let mut group = c.benchmark_group("In memory");
+    for scenario in TEST_SCENARIOS {
+        group.bench_with_input(
+            BenchmarkId::new("is_rate_limited", scenario),
+            scenario,
+            |b: &mut Bencher, test_scenario: &&TestScenario| {
+                let storage = Box::new(InMemoryStorage::default());
+                bench_is_rate_limited(b, test_scenario, storage);
+            },
+        );
+        group.bench_with_input(
+            BenchmarkId::new("update_counters", scenario),
+            scenario,
+            |b: &mut Bencher, test_scenario: &&TestScenario| {
+                let storage = Box::new(InMemoryStorage::default());
+                bench_update_counters(b, test_scenario, storage);
+            },
+        );
+        group.bench_with_input(
+            BenchmarkId::new("check_rate_limited_and_update", scenario),
+            scenario,
+            |b: &mut Bencher, test_scenario: &&TestScenario| {
+                let storage = Box::new(InMemoryStorage::default());
+                bench_check_rate_limited_and_update(b, test_scenario, storage);
+            },
+        );
+    }
+    group.finish();
 }
 
-fn bench_check_rate_limited_and_update_redis(c: &mut Criterion) {
-    c.bench_function_over_inputs(
-        &format!("check_rate_limited_and_update (redis)"),
-        |b: &mut Bencher, test_scenario: &&TestScenario| {
-            let storage = Box::new(RedisStorage::default());
-            bench_check_rate_limited_and_update(b, test_scenario, storage);
-        },
-        TEST_SCENARIOS.to_vec(),
-    );
+#[cfg(feature = "redis")]
+fn bench_redis(c: &mut Criterion) {
```

Comment:
`bench_redis` is pretty much the same as `bench_in_mem` except for the storage. Isn't there a way to generalize into

```rust
fn bench_generic(c: &mut Criterion, storage: Box<dyn CounterStorage>)
```

?

Reply:
@eguzki I actually did something like that (or almost, see below), but thought it might be less readable… So decided to drop the change. I'll let you judge of which version is better, see this change set: dc79ae5

Let me know whether to keep it (I'd squash it before merging) or whether to just drop it…

Reply:
It is ok as it is. It is a tests and readability matters.